source/Mlos.Python/mlos/Optimizers/BayesianOptimizer.py

```diff
@@ -102,10 +102,14 @@ def get_all_observations(self):
         return self._feature_values_df.copy(), self._target_values_df.copy()
 
     @trace()
-    def suggest(self, random=False, context=None):
+    def suggest(self, random=False, context: Point = None):
         # TODO: pass context to the suggest method
+        if self.optimization_problem.context_space is not None:
+            if context is None:

```

Comment:
This makes context a required parameter whenever the context space is setup.  Is that desired/necessary?
Static context may not change in between suggestions and seems unnecessary/expensive to provide each time.
Dynamic context may also be expensive to collect each time.
Maybe the answer in both cases is just a cache?

Reply:
I hadn't really thought about the grpc case. When using the Python interface this doesn't make a difference, right?
If we have remote agents, then there could be multiple agents with different contexts that each don't change, but we would need to keep track of which agent corresponds to which context. Which I guess is the same as a cache?

Reply:
You bring up a good point about having to map "saved context" to the appropriate "session" (agent instance).
Would we need that in a local python case as well?  (e.g. multi-process tests)

Reply:
Well all the information should be encoded in the context, right? And what's not distinguished by context should be shared if you connect to the same optimizer.

Reply:
From other discussions, I think the intention for this PR is to only handle context for the local case, and address the remote case later on, so I think this conversation can be closed for now.